luno_python/stream_client.py

```
@@ -0,0 +1,201 @@
+        self._status = status_update['status']
+
+
+async def _read_from_websocket(ws, pair: Pair, update_f: StateUpdate):
+    state = None
+    is_first = True
+
+    async for message in ws:
+        try:
+            body = json.loads(message)
+        except ValueError:
+            raise Exception(message)
```

Ruff (0.15.6)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)